syntax = "proto3";

package amocrm_connect_service;

service AmocrmConnectService {
  rpc TryConnect (AmocrmConnectRequest) returns (AmocrmConnectResponse);
  rpc GetInfo (GetInfoRequest) returns (GetInfoResponse);
}



message AmocrmConnectRequest {
  string login = 1;
  string password = 2;
  string host = 3;
}

message GetInfoRequest {
  string email = 1;
  string password = 2;
  string host = 3;
}

message Data {
  string message = 1;
}

message AmocrmConnectResponse {
  string answer = 1;
  bool success = 2;
  Data data = 3;
  float execution = 4;
}

message GetInfoResponse {
  repeated Pipeline pipelines = 1;
  repeated Field fields = 2;
}


message Pipeline {
  int32 id = 1;
  string name = 2;
  int32 sort = 3;
  repeated Status statuses = 4;
}

message Status {
  int32 id = 1;
  int32 sort = 2;
  string name = 3;
}

message Field {
  int32 id = 1;
  string name = 2;
  string type = 3;
  string active_value = 4;
  repeated Select possible_values = 5;
}

message Select {
  int32 id = 1;
  string value = 2;
  int32 sort = 3;
}

message Chat {
  string chat_id = 1;
  string message = 2;
  string id = 3;
  int32 pipeline_id = 4;
  int32 lead_id = 5;
  int32 status_id = 6;
  string messages_history = 7;
}
